src/compiler/scala/tools/nsc/javac/JavaParsers.scala

```diff
@@ -135,6 +132,18 @@ trait JavaParsers extends ast.parser.ParsersCommon with JavaScanners {
       DefDef(Modifiers(Flags.JAVA), nme.CONSTRUCTOR, List(), List(vparams), TypeTree(), blankExpr)
     }
 
+    def withConstructor(stats: List[Tree]): List[Tree] = {
+      val pureStats = stats map {
+        case DocDef(_, defn) => defn
+        case defn => defn
+      }
+      if(treeInfo.firstConstructor(pureStats) == EmptyTree) {
```

Comment:
Move the `DocDef` unwrapping into `override def firstConstructor` in `scala.tools.nsc.ast.TreeInfo` to keep te `DocDef` special cases in one place?